core/tee/tee_rpmb_fs.c

```diff
@@ -1112,7 +1112,7 @@ static TEE_Result tee_rpmb_init(uint16_t dev_id)
 
 		res = tee_rpmb_verify_key_sync_counter(dev_id);
 		if (res == TEE_ERROR_ITEM_NOT_FOUND &&
-			!rpmb_ctx->key_verified) {
+			!rpmb_ctx->key_verified && plat_rpmb_ready()) {
```

Comment:
This happens after the key already has been generated. Wouldn't it be enough to return an error from `tee_otp_get_hw_unique_key()`?

Reply:
This will also make it impossible to use with the REE FS, or am I wrong? I can see that one might want to use REE FS for development.

Reply:
This will also make it impossible to use the REE FS, since the initialization will fail as well, or am I wrong?

Reply:
`tee_otp_get_hw_unique_key()` failing with make a lot of things stop working.

Reply:
In that case I'd like to not break OP-TEE for open devices and prefer the solution in this PR. Subkey derivation will still work with open devices, but will use a test key and not the One Time Programmable Master Key (OTPMK). We should still complain loudly that this is insecure, but should allow it for development. This PR strives to prevent the irreversible write of the RPMB key, since the key cannot be changed after the write.